hack/runtime-migrator/cmd/backup/main.go

```diff
@@ -24,21 +22,34 @@ func main() {
 
 	gardenerNamespace := fmt.Sprintf("garden-%s", cfg.GardenerProjectName)
 
-	_, err := config.SetupKubernetesKubeconfigProvider(cfg.GardenerKubeconfigPath, gardenerNamespace, expirationTime)
+	kubeconfigProvider, err := config.SetupKubernetesKubeconfigProvider(cfg.GardenerKubeconfigPath, gardenerNamespace, expirationTime)
 	if err != nil {
 		slog.Error(fmt.Sprintf("Failed to create kubeconfig provider: %v", err))
 		os.Exit(1)
 	}
 
-	_, err = config.CreateKcpClient(&cfg)
+	shootClient, err := config.SetupGardenerShootClient(cfg.GardenerKubeconfigPath, gardenerNamespace)
 	if err != nil {
-		slog.Error("Failed to create kcp client", slog.Any("error", err))
+		slog.Error("Failed to setup Gardener shoot client", slog.Any("error", err))
 		os.Exit(1)
 	}
 
-	_, err = config.SetupGardenerShootClient(cfg.GardenerKubeconfigPath, gardenerNamespace)
+	backup, err := NewBackup(cfg, kubeconfigProvider, shootClient)
 	if err != nil {
-		slog.Error("Failed to setup Gardener shoot client", slog.Any("error", err))
+		slog.Error("Failed to initialize backup", slog.Any("error", err))
+		os.Exit(1)
+	}
+
+	slog.Info("Reading runtimeIds from input file")
+	runtimeIds, err := config.GetRuntimeIDsFromInputFile(cfg)
+	if err != nil {
+		slog.Error("Failed to read runtime Ids from input", slog.Any("error", err))
+		os.Exit(1)
+	}
+
+	err = backup.Do(context.Background(), runtimeIds)
```

Comment:
Please consider using context with a timeout.

Reply:
I don't see a point in adding timeout here. We don't need a timeout for the whole backup process. The context is passed to `Do` method, and later used for adding timeout to k8s operations.